frontend/jupyter/src/app/pages/form/form-default/form-image/form-image.component.ts

```diff
@@ -145,23 +145,26 @@ export class FormImageComponent implements OnInit, OnDestroy {
     return displayName;
   }
 
-  nonEmployeeUsers(imageGroup: Map<string, string>): boolean{
+  shouldEnable(imageGroup: Map<string, string>): boolean{
     if (imageGroup == null) {
-      this.disabled = false;
-      return false;
-    }
-    //const label = Object.keys(this.enabledCondition["labels"])[0];
-    //const labelValue = this.enabledCondition["labels"][label];
-    //this.nsinfo.metadata.labels["state.aaw.statcan.gc.ca/non-employee-users"]
-    else {
-      this.disabled = true;
+      this.enabled = true;
       return true;
     }
-  }
-
-  test(imageGroup: Map<string, string>): [string, any][]{
-    const att = Object.entries(imageGroup);   
-    return att;
+    const conditionLabels = Object.keys(imageGroup["labels"]);
+    const metadataLabels = Object.keys(this.nsinfo.metadata.labels);
+    var enable = true;
+    for (const label of conditionLabels) {
+      if (metadataLabels.includes(label)) {
+        if (this.nsinfo.metadata.labels[label] !=  imageGroup["labels"][label]) {
+          enable = false;
+        }
+      }
+      else {
+        enable = false;
+      }
+    }
+    this.enabled = enable;
+    return enable;
```

Comment:
This looks like it's following a "one return statement per function" rule that some university professors teach. Don't listen to those professors; they are wrong. :stuck_out_tongue: